packages/pangea-sdk/pangea/verify_audit.py

```diff
@@ -127,22 +148,51 @@ def _verify_unpublished_membership_proof(root_hash, node_hash: str, proof: Optio
     return succeeded
 
 
-def _verify_membership_proof(tree_name: str, tree_size: int, node_hash: str, proof: Optional[str]) -> Optional[bool]:
+def _fetch_roots(tree_name: str, tree_size: int, leaf_index: Optional[int]) -> Optional[bool]:
+    global pub_roots
+
+    log_section("Fetching published roots from Arweave")
+
+    succeeded = None
+    needed_roots = {tree_size}
+    if leaf_index:
+        needed_roots |= {leaf_index, leaf_index + 1}
+    pending_roots = needed_roots - set(pub_roots.keys())
+
+    try:
+        if pending_roots:
+            pub_roots |= {int(k): v for k, v in get_arweave_published_roots(tree_name, pending_roots).items()}  # type: ignore[operator]
+            pending_roots = needed_roots - set(pub_roots.keys())
+        succeeded = True
+        if pending_roots and audit:
+            logger.debug("Published root could not be fetched from Arweave")
+            logger.debug("Fetching published roots from Pangea")
+            pub_roots |= {int(k): v for k, v in _get_roots(pending_roots).items()}  # type: ignore[operator]
+            pending_roots = needed_roots - set(pub_roots.keys())
+            succeeded = None
+        if pending_roots:
+            raise ValueError("Published root could not be fetched")
+    except:
+        succeeded = False
+
+    log_result("Fetching published roots", succeeded)
+    logger.info("")
```

Comment:
Why is this?

Reply:
the empty log? I guess it's for flushing the output